softlearning/environments/adapters/dm_control_adapter.py

```diff
@@ -0,0 +1,144 @@
+"""Implements an adapter for DeepMind Control Suite environments."""
+
+from collections import OrderedDict
+
+import numpy as np
+from dm_control import suite
+from dm_control.rl.specs import ArraySpec, BoundedArraySpec
+from gym import spaces
+
+from .softlearning_env import SoftlearningEnv
+
+
+# TODO(hartikainen): Add information of the available environments.
+# See `gym_adapter.py` for example.
+DM_CONTROL_ENVIRONMENTS = {}
+
+
+def convert_dm_control_to_gym_space(dm_control_space):
+    if isinstance(dm_control_space, BoundedArraySpec):
```

Comment:
Can you leave a note here saying that the order of these if-statements matter.

Reply:
Or alternatively raise in the below if-statement if we ever end up there with `BoundedArraySpec`.

Reply:
I wrote a comment of the form: 
`# Note: Need to check (1) whether dm_control_space is a BoundedArraySpec
    # and then (2) whether dm_control_space is an ArraySpec.
    # The order of these if conditions matters:
    # generally, dm_control observation_specs are of type ArraySpec
    # generally, dm_control action_specs are of type BoundedArraySpec`

Not sure about where to raise that `BoundedArraySpec` error?

Reply:
I think this is good. Let's format this to be a [docstring](https://www.python.org/dev/peps/pep-0257/). I.e.

```python
def convert_dm_control_to_gym_space(dm_control_space):
    """Recursively convert dm_control space into gym space.

    Note: Need to check the following cases of the input type, in the following 
    order:
       (1) BoundedArraySpec
       (2) ArraySpec
       (3) OrderedDict.
    Generally, dm_control observation_specs are OrderedDict with other spaces 
    (e.g. ArraySpec) nested in it.
    Generally, dm_control action_specs are of type `BoundedArraySpec`.
    To handle dm_control observation_specs as inputs, we check the following 
    input types in order to enable recursive calling on each nested item.
    """
```

Reply:
Re raising the error: We could've made the ordering explicit by modifying the second if-statement as follows:
```python
    elif isinstance(dm_control_space, ArraySpec):
        if isinstance(dm_control_space, BoundedArraySpec):
            raise ValueError("The order of the if-statements matter.")
        return spaces.Box(
            low=-float("inf"),
            high=float("inf"),
            shape=dm_control_space.shape,
            dtype=dm_control_space.dtype)
```

The idea is that if someone in the future needs to make changes in the code, they will get an error if the order of the if-statements is changed. This would work even if the comments are ignored.

Reply:
Yep, makes sense. Committed.